persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -519,6 +518,19 @@ protected void loginInternally(io.stargate.db.AuthenticatedUser user) {
                         ClientWarn.instance.resetWarnings();
                       }
                     })
+                .doOnSubscribe(
+                    disposable -> {
+                      // When running inside DSE, query tasks clear ExecutorLocals before
+                      // running, which is handled by its Message.channelRead0. In Stargate
+                      // requests may come from Epoll threads of the CQL module, from HTTP
+                      // threads or any other client. So here we ensure that DSE code starts
+                      // processing the request with a clean thread local state.
+                      ExecutorLocals.set(null);
```

Comment:
OK, I don't know where and how is `ExecutorLocals` being actually used.. But I understand the problem, you actually wanna clear state one a thread that picks up the request execution I guess.. 

First problem is that there is absolutely no-guarantee that thread that does `doOnSubscribe` also does the stuff where the ThreadLocal is used. Seems that the execution is actually outside of this class, thus in the creation of the reactive sequence you never know which schedulers are used. But if it works, it could be a quick fix.

The reactive stuff and thread locals are not really friends. This is why usually you wanna handle these sort of things with reactive context. Reactive context is passed along and you always have the guarantee that you actually have correct context at the time of the execution, no matter which thread it is ending in. But of course, there has to be a "contract" between two sides and yea this old style thread local needs to go away. I know this is not so simple :)

Anyway, I did wrote a quick blog on similar matter (https://www.novatec-gmbh.de/en/blog/how-can-the-mdc-context-be-used-in-the-reactive-spring-applications/), yes it was Kotlin and different stuff, but a similar idea could be applied here as well.. Although a way nicer things, as we anyway are in control of the `ExecutorLocals`, would be to remove it completely and migrate to context. In the blog, we needed to do it, as we were depending on the 3rd party impl.

Reply:
Yes, this is tricky indeed. The propagation context data in RxJava in this case is actually handled by `TPCRunnable`, which is part of DSE code. It picks up `ExecutorLocals` from current thread and moves them to another thread in case Rx tasks switch threads.

Reply:
@ivansenic,

> First problem is that there is absolutely no-guarantee that thread that does doOnSubscribe also does the stuff where the ThreadLocal is used.

That is correct, more specifically `doOnSubscribe()` is guaranteed to be called by the thread setting up the reactive chain, regardless of where the subscription ends up being handled, but that's what we care about in this ticket: the rest, that is the actual propagation, as mentioned by @dimas-b, is handled by the TPC framework.